### PR TITLE
Refactor code quality issues

### PR DESCRIPTION
**Pull request by akshgpt7**
## Description

Hi :wave:, I ran [DeepSource](https://deepsource.io/) analysis on my fork of the repo, and found some interesting [code quality and performance issues](https://deepsource.io/dashboard/) that can improve the general code quality and performance here.
This PR fixes a few of the issues detected.

## Summary of changes

- Remove unnecessary `len()` call for comparison check.
- Refactor unnecessary use of comprehension.
- Use literals instead of function calls to improve performance.
- Use `sys.exit()` calls instead of bare `exit()`.
- Remove unnecessary import aliases.
- Add `.deepsource.toml` file for continuous analysis on bug risks/performance/code-quality issues on new changes.

### CODE REVIEW

File: .deepsource.toml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+version = 1
+
+test_patterns = ["tests/**"]
+
+exclude_patterns = ["docker/**"]
```
Comment (ahopkins) [inline .deepsource.toml:5]: What about the docs folder?

**Review by ahopkins — APPROVED** [review]
Nice, this looks like a good change. I am not familiar with this tool. Is this something that we could add to `tox` and then run with CI?

**Comment by akshgpt7** [comment]
@ahopkins 
> Is this something that we could add to `tox` and then run with CI?

DeepSource is a static analysis tool (which runs separately) and not exactly for testing, so it can't be integrated directly with `tox`. However, you can activate analysis for your repository to continuously analyze and detect bug-risks/performance issues/anti-patterns for each new change introduced. It significantly reduces code review time, and you can also use the Autofix feature to fix them with one click.

If you'd like, here's how you can quickly activate DeepSource to continuously analyze your repository:
- Sign up on [DeepSource](https://deepsource.io/signup) and activate analysis for your repository.
- Create `.deepsource.toml` configuration which you can use to configure your analysis settings (This PR already added that, but you can edit it any time).
- Activate/Check your analysis [here](https://deepsource.io/gh/sanic-org/sanic).

If you have any doubts or questions, you can check out the [docs](https://deepsource.io/docs), or feel free to reach out :)

File: .deepsource.toml (hunk 1 above)
Comment (akshgpt7) [reply .deepsource.toml:5]: That can be excluded too, but I saw the `conf.py` file in there, so I didn't add the docs folder here.

**Comment by ahopkins** [comment]
Thanks, I'll look into that over the weekend. These changes seemed logical enough, but I would not want autofix enabled.

**Comment by akshgpt7** [comment]
It's not _enabled_ on autopilot. You just have the additional choice to Autofix some detected issues, or manually fix them, or ignore them altogether. Even if you choose to Autofix a particular issue, it opens a separate PR, rather than directly commiting the changes, so nothing goes through undetected :)

**Comment by akshgpt7** [comment]
@ahopkins, Just making sure if you need any help in activating analysis?
I ask so because you merged the `.deepsource.toml` file, which will only work if analysis is activated for that repository (through the steps mentioned above).

**Comment by akshgpt7** [comment]
@ahopkins just let me know if you need any help in activating :)

**Pull request merged**